src/components/MapView/Download/index.tsx

```diff
@@ -0,0 +1,197 @@
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  Theme,
+  Typography,
+  WithStyles,
+  withStyles,
+} from '@material-ui/core';
+import Menu, { MenuProps } from '@material-ui/core/Menu';
+
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import {
+  faFileExport,
+  faCaretDown,
+  faImage,
+} from '@fortawesome/free-solid-svg-icons';
+
+import { jsPDF } from 'jspdf';
+
+import { useSelector } from 'react-redux';
+import { mapSelector } from '../../../context/mapStateSlice/selectors';
+
+const ExportMenu = withStyles((theme: Theme) => ({
+  paper: {
+    border: '1px solid #d3d4d5',
+    backgroundColor: theme.palette.primary.main,
+  },
+}))((props: MenuProps) => (
+  <Menu
+    elevation={0}
+    getContentAnchorEl={null}
+    anchorOrigin={{
+      vertical: 'bottom',
+      horizontal: 'center',
+    }}
+    transformOrigin={{
+      vertical: 'top',
+      horizontal: 'center',
+    }}
+    {...props}
+  />
+));
+
+const ExportMenuItem = withStyles((theme: Theme) => ({
+  root: {
+    color: theme.palette.common.white,
+  },
+}))(MenuItem);
+
+function Download({ classes }: DownloadProps) {
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+  const [open, setOpen] = useState(false);
+  const selectedMap = useSelector(mapSelector);
+  const previewRef = useRef<HTMLCanvasElement>(null);
+
+  const handleClick = (event: React.MouseEvent<HTMLElement>) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  useEffect(() => {
+    if (open && selectedMap) {
+      const activeLayers = selectedMap.getCanvas();
+      const canvas = previewRef.current;
+      canvas!.setAttribute('width', activeLayers.width.toString());
+      canvas!.setAttribute('height', activeLayers.height.toString());
+      const context = canvas!.getContext('2d');
+      context!.drawImage(activeLayers, 0, 0);
+    }
+  });
+
+  const modalClose = () => {
+    setOpen(false);
+  };
+
+  const download = (format: String) => {
+    const ext = format === 'pdf' ? 'png' : format;
+    const canvas = previewRef!.current;
+    const file = canvas!.toDataURL(`image/${ext}`);
+    if (format === 'pdf') {
+      // eslint-disable-next-line new-cap
+      const pdf = new jsPDF({
+        orientation: 'landscape',
+      });
+      const imgProps = pdf.getImageProperties(file);
+      const pdfWidth = pdf.internal.pageSize.getWidth();
+      const pdfHeight = (imgProps.height * pdfWidth) / imgProps.width;
+      pdf.addImage(file, 'PNG', 0, 0, pdfWidth, pdfHeight);
+      pdf.save('map.pdf');
+    } else {
+      const link = document.createElement('a');
+      link.setAttribute('href', file);
+      link.setAttribute('download', `map.${ext}`);
+      link.click();
+    }
+    setOpen(false);
+  };
+
+  return (
+    <Grid item>
+      <Button variant="contained" color="primary" onClick={handleClick}>
+        <FontAwesomeIcon style={{ fontSize: '1.2em' }} icon={faFileExport} />
```

Comment:
Maybe we can use MUI icons for better theme consistency? We can use FontAwesome only if the icon doesn't exist in MUI's library.

Reply:
I tried to update using MUI icons for the top buttons: 
![image](https://user-images.githubusercontent.com/25983260/113530871-6212fa80-95f1-11eb-974d-22f09ec30d52.png)

but got an error when i run yarn test
![image](https://user-images.githubusercontent.com/25983260/113530946-95558980-95f1-11eb-945e-82c124b92d7c.png)

do i need to update setupTests or something?

Reply:
hm, maybe you can mock the MUI icon with a simple string like `jest.mock('icon path', ()=>'<ChevronArrow/>')` 
If that fails, then we can just use the FontAwesome icon (they are both similar anyway)

Reply:
i've tried using like `jest.mock('@material-ui/icons/Visibility', () => 'mock-visibility');`, all tests are passed after i updated the snapshot, but the error message above still remaining.
so, use FontAwesome instead?

Reply:
Did you trying a similar implementation as https://github.com/WFP-VAM/prism-frontend/blob/master/src/setupTests.tsx ?

Reply:
i installed `@material-ui/icons` and added `jest.mock` for specific test file where i used the icon. i think it similar with the implementation in `setupTest`.

i just remembered, do we need to include the `Accordion` component to the list in `setupTest`?

Reply:
You could spot which lib components take multiple lines and mock them in `setuptTest`

Another thing you can do is break down your component into a few different ones, and test them separately. So that you only test the component within the `.map` once. @Ry-DS any other ideas?

Reply:
You don't need to mock Accordion if it work fine without the mock imo (we want to capture the children and mocking prevents that). But if you gave errors, or the snapshot is immensely large because of Accordion, we can mock it :)